contrib/auto-render/auto-render.js

```diff
@@ -83,30 +85,26 @@ const defaultOptions = {
     ],
 };
 
-const extend = function(obj) {
-    // Adapted from underscore.js' `_.extend`. See LICENSE.txt for license.
-    let source;
-    let prop;
-    const length = arguments.length;
-    for (let i = 1; i < length; i++) {
-        source = arguments[i];
-        for (prop in source) {
-            if (Object.prototype.hasOwnProperty.call(source, prop)) {
-                obj[prop] = source[prop];
-            }
-        }
-    }
-    return obj;
-};
+function AutoRenderSettings(options) {
+    // allow null options
+    options = options || {};
+    this.delimiters = options.delimiters || defaultOptions.delimiters;
+    this.ignoredTags = options.ignoredTags || defaultOptions.ignoredTags;
+}
 
 const renderMathInElement = function(elem, options) {
     if (!elem) {
         throw new Error("No element provided to render");
     }
 
-    options = extend({}, defaultOptions, options);
+    const autoRenderOptions = new AutoRenderSettings(options);
```

Comment:
I think this logic might work instead:

```js
const optionsCopy = Object.assign({}, defaultOptions, options);
```

[`defaultOptions`](https://github.com/alexpearce/KaTeX/blob/8a11755ac12899816c30f73e13d952f740d74184/contrib/auto-render/auto-render.js#L76-L88) is merged first (probably should rename this to `defaultAutoRenderOptions`) and then the user `options` is merged. So if `options` contains the same keys as `defaultOptions`, the former will override the latter. Then indeed there's no need for `AutoRenderSettings`.

What do you think?